profiles/audio/avdtp.c

```diff
@@ -3429,6 +3429,11 @@ int avdtp_discover(struct avdtp *session, avdtp_discover_cb_t cb,
 	if (err == 0) {
 		session->discover->cb = cb;
 		session->discover->user_data = user_data;
+	} else {
```

Comment:
You can use something like:
`else if (session->discover)`

Anyway, please sense this change to the list instead.